### ISSUE TITLE
ImportError: No module named 'kiwisolver' 

### ISSUE THREAD

**Issue by karshan77** [nucleic/kiwi#68]
Hi,

I am getting the error when I'm executing the command `python3 legacy/train.py --logtostderr --train_dir=training/ --pipeline_config_path=training/ssd_mobilenet_v1_pets.config`.

So, I have tried to install kiwisolver by following command `pip3 install kiwisolver`. Even though the error not getting resolved. So, I went checked into the kiwisolver installed directory. Only one directory is available in the name "kiwisolver-1.1.0.dist-info". No kiwisolver libraries has been installed. How to install kiwisolver properly?

I am working on ubuntu 16.04.

Thanks.

**Comment by MatthieuDartiailh** [comment]
Could you please post the full error message and the output of pip ? Running pip to install kiwisolver should be enough. As a side note it is usually better to avoid messing with your distro Python distribution by using virtual environment.

**Comment by karshan77** [comment]
Hi,

This is the full error:
```
Traceback (most recent call last):
  File "legacy/train.py", line 51, in <module>
    from object_detection.builders import model_builder
  File "/usr/local/lib/python3.5/dist-packages/object_detection-0.1-py3.5.egg/object_detection/builders/model_builder.py", line 34, in <module>
    from object_detection.meta_architectures import ssd_meta_arch
  File "/usr/local/lib/python3.5/dist-packages/object_detection-0.1-py3.5.egg/object_detection/meta_architectures/ssd_meta_arch.py", line 30, in <module>
    from object_detection.utils import visualization_utils
  File "/usr/local/lib/python3.5/dist-packages/object_detection-0.1-py3.5.egg/object_detection/utils/visualization_utils.py", line 27, in <module>
    import matplotlib.pyplot as plt  # pylint: disable=g-import-not-at-top
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/pyplot.py", line 32, in <module>
    import matplotlib.colorbar
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/colorbar.py", line 34, in <module>
    import matplotlib.gridspec as gridspec
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/gridspec.py", line 25, in <module>
    import matplotlib._layoutbox as layoutbox
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/_layoutbox.py", line 19, in <module>
    import kiwisolver as kiwi
ImportError: No module named 'kiwisolver'

```

Output message when installing kiwisolver:
`pip3 install kiwisolver`

```
Collecting kiwisolver
  Using cached https://files.pythonhosted.org/packages/ee/18/4cd2e84c6aff0c6a50479118083d20b9e676e5175a913c0ea76d700fc244/kiwisolver-1.1.0-cp35-cp35m-manylinux1_x86_64.whl
Requirement already satisfied: setuptools in /home/shankar/Downloads/envs/lib/python3.5/site-packages (from kiwisolver) (41.0.1)
Installing collected packages: kiwisolver
Successfully installed kiwisolver-1.1.0
```
I have tried uninstalling and installed it once again. It is also not working.

Thanks.

**Comment by MatthieuDartiailh** [comment]
Kiwi is implemented using C++ only (no python file) as a consequence you should only see a kiwisolver.***.so in site-packages in addition of the dist-info folder. Could you check whether or not it is present ?
The wheel are tested as part of the build process and I locally checked on MacOS that the python 3.5 wheel work. I will spin up a VM to check also on linux. In the meantime if you have a c++ compiler installed, you can try to install from source directly: `pip3 install https://github.com/nucleic/kiwi/tarball/master/`

**Comment by MatthieuDartiailh** [comment]
I just tested on a fresh Docker image of ubuntu 16 and I can install without any issue on Python 3.5. Could there be something (like a folder named kiwisolver) on your path that would shadow the installed package ?

**Comment by karshan77** [comment]
Hi @MatthieuDartiailh ,

I have installed as you have mentioned but same thing happened. But, I have noticed ont thing. Kiwisolver has been installed differently in my system. I don't know why. There are two things getting installed:

1. kiwisolver-1.1.0.dist-info
2. kiwisolver.cpython-35m-x86_64-linux-gnu.so (I haven't noticed this file has been installed)

I have copied this file into my Python path directory then I'm not getting the error. First time I'm seeing this file getting installed using pip command.

Thank you so much.

**Comment by MatthieuDartiailh** [comment]
This is the expected file (once again this is related to the fact that kiwisolver is a single module implemented in C). I am wondering why site-packages is not on the path... but I am happy things work again for you.

**Comment by shresthadk51** [comment]
I did have same issue ModuleNotFoundError: No module named 'kiwisolver' - what I found is that kiwisolver-1.1.0.dist-info was installed on ...AppData\Roaming\Python\Python37\site-packages. I deleted the entire folder kiwisolver-1.1.0.dist-info and reinstalled with pip3 install kiwisolver. Now it is installed on ...\appdata\local\programs\python\python37\lib\site-packages and tested the code it passed!
